geode-core/src/main/java/org/apache/geode/cache/persistence/PersistentID.java

```diff
@@ -58,4 +58,11 @@ public interface PersistentID extends DataSerializable {
    * @since GemFire 7.0
    */
   UUID getUUID();
+
+  /**
+   * The failed reason for the persistent
+   *
+   * @return the faild reason code for the persistent
+   */
+  int getFailedReason();
```

Comment:
Hi @albertogpz ,
I also have concern about it when adding this getFailedReason() , I agree with your point.
And do you think it is good for the faildReason in  PersistentMemberPattern.java,  as this class is also not for backup. If it is also not good, we will need to modify all the interfaces for backup, like BackupService.prepareBackup(), BackupService.doBackup() as they all return only the HashSet<PersistentID>.

And if we agree the faildReason in PersistentMemberPattern.java, it will much simple, only remove the getFailedReason() in PersistentID.java, and covert the PersistentId to PersistentMemberPattern when need to get failed reason.